app/models/manageiq/providers/microsoft/infra_manager/refresh_parser.rb

```diff
@@ -193,11 +198,36 @@ def process_virtual_switches(host)
     end
 
     def process_logical_networks(logical_networks)
-      logical_networks.collect do |ln|
-        {
+      network_hashes = []
+
+      logical_networks.each do |ln|
+        result = {
           :name    => ln['Name'],
           :uid_ems => ln['ID'],
         }
+
+        network_hashes << result
+
+        Array(ln['VMNetworks']).each do |vm_network|
+          network_hashes << {
+            :name    => vm_network['Name'],
+            :uid_ems => vm_network['ID'],
+            :subnets => process_subnets(vm_network),
+          }
+        end
+      end
+
+      network_hashes
+    end
+
+    def process_subnets(vm_network)
+      subnets = Array(vm_network['VMSubnet'])
+      subnets.map do |subnet|
+        {
+          :name    => subnet['Name'],
+          :ems_ref => subnet['ID'],
+          :cidr    => subnet['SubnetVLans'],
```

Comment:
@djberg96 is `SubnetVLans` a valid CIDR?

I get `["192.168.1.0/24-0", "230", "192.168.31.0/24-231", "241", "231", "192.168.32.0/24-241", "192.168.1.0/24-0", "230", "192.168.31.0/24-231", "241", "231", "192.168.32.0/24-241", "192.168.1.0/24-0"]` which looks like a combination of (I assume) `cidr-vlanid` and just `vlanid`

Reply:
@agrare Yeah, I was confused by that as well when looking at the raw powershell objects. Let me find a few and paste them here.

Reply:
Looks like we're going to have to go a layer deeper:
```
PS C:\> $vmnetwork = Get-SCVMNetwork -Name 'dberger'

PS C:\> $vmnetwork

Name                    : dberger
LogicalNetwork          : rhq
VMSubnet                : {dbergersubnet}
VMNetworkGateways       : {}
VPNConnections          : {}
NATConnections          : {}
RoutingDomainId         : abcdef
IsolationType           : WindowsNetworkVirtualization
UseGRE                  : True
PAIPAddressPoolType     : IPV4
CAIPAddressPoolType     : IPV4
ExternalName            :
NetworkEntityAccessType : VmmManaged
IsAssigned              : True
IsPrivateVlan           : False
HasGatewayConnection    : False
NetworkManager          :
PortACL                 :
GrantedToList           : {}
UserRoleID              : xyz
UserRole                : Administrator
Owner                   : SomeDomain\Administrator
ObjectType              : VMNetwork
Accessibility           : Public
IsViewOnly              : False
Description             : Test VM Network
AddedTime               : 10/9/2017 4:57:36 PM
ModifiedTime            : 8/15/2017 12:21:19 PM
Enabled                 : True
MostRecentTask          :
ServerConnection        : Microsoft.SystemCenter.VirtualMachineManager.Remoting.ServerConnection
ID                      : xxx
MarkedForDeletion       : False
IsFullyCached           : True
MostRecentTaskIfLocal   :

PS C:\> $vmnetwork.vmsubnet

Name                           : dbergersubnet
Description                    :
VMSubnetID                     : 1234567
VMNetwork                      : dberger
SubnetVLans                    : {192.168.1.0/24-0}
AllowsIntraPortCommunication   : True
MaxNumberOfPorts               :
VirtualSwitchExtensionManager  :
LogicalNetworkDefinition       :
ExternalId                     : abc
NetworkEntityAccessType        : VmmManaged
ExternalSyncTime               :
ExternalVMNetworkIsolationType :
ExternalVMSubnetIdentifier     :
IsVMMInternal                  : False
PortACL                        :
ServerConnection               : Microsoft.SystemCenter.VirtualMachineManager.Remoting.ServerConnection
ID                             : xxx
IsViewOnly                     : False
ObjectType                     : VMSubnet
MarkedForDeletion              : False
IsFullyCached                  : True



PS C:\> $vmnetwork.vmsubnet.subnetvlans


Subnet                 : 192.168.1.0/24
VLanID                 : 0
SecondaryVLanID        : 0
IsVLanEnabled          : False
IsSecondaryVLanEnabled : False
SupportsDHCP           : True
ID                     :
IsAssignedToVMSubnet   : True

A number like "240" is just a vlan ID and could be skipped, so I guess we'd have to check the format as well.